sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuBatchUtils.scala

```diff
@@ -63,6 +63,30 @@ object GpuBatchUtils {
     estimateGpuMemory(field.dataType, field.nullable, rowCount)
   }
 
+  /**
+   * Get the minimum size a column could be that matches these conditions.
+   */
+  def minGpuMemory(dataType:DataType, nullable: Boolean, rowCount: Long): Long = {
+    val validityBufferSize = if (nullable) {
+      calculateValidityBufferSize(rowCount)
+    } else {
+      0
+    }
+
+    val dataSize = dataType match {
+      case DataTypes.BinaryType | DataTypes.StringType | _: MapType | _: ArrayType=>
+        // No data, just offsets
+        calculateOffsetBufferSize(rowCount)
```

Comment:
It isn't clear to me why we just measure offsets for these types. Could you expand the comment here to explain?